sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeActionsAsyncClient.java

```diff
@@ -165,16 +203,46 @@ PollerFlux<AnalyzeActionsOperationDetail, AnalyzeActionsResultPagedFlux> beginAn
             options = getNotNullAnalyzeActionsOptions(options);
             final Context finalContext = getNotNullContext(context)
                                              .addData(AZ_TRACING_NAMESPACE_KEY, COGNITIVE_TRACING_NAMESPACE_VALUE);
+            final boolean finalIncludeStatistics = options.isIncludeStatistics();
+
+            if (service != null) {
+                final AnalyzeTextJobsInput analyzeTextJobsInput =
```

Comment:
I believe this part creates a poller but I wonder it could be abstracted over the service calls and shared between both service APIs. For example, can we write a function, let's name it createPoller or something, that takes two callbacks as input where the first sends the POST request and the second sends the GET request? and then use this function to handle both service API calls?

Reply:
Create an issue to address the abstraction in the next PR: https://github.com/Azure/azure-sdk-for-java/issues/28835